python/pyspark/sql/pandas/serializers.py

```diff
@@ -420,7 +421,11 @@ def __init__(
     def arrow_to_pandas(self, arrow_column):
         import pyarrow.types as types
 
-        if self._df_for_struct and types.is_struct(arrow_column.type):
+        if (
+            self._df_for_struct
+            and types.is_struct(arrow_column.type)
+            and not is_variant(arrow_column.type)
+        ):
             import pandas as pd
```

Comment:
Why do we want to avoid this path for variant types? Maybe we should leave a comment explaining it.